src/resolver.ts

```diff
@@ -3644,15 +3692,28 @@ export class Resolver extends DiagnosticEmitter {
 
     // Resolve type arguments if generic
     if (prototype.is(CommonFlags.Generic)) {
-      resolvedTypeArguments = this.resolveTypeArguments( // reports
-        assert(prototype.typeParameterNodes), // must be present if generic
-        typeArgumentNodes,
-        flow,
-        ctxElement,
-        ctxTypes, // update
-        reportNode,
-        reportMode
-      );
+
+      // if no type arguments are provided, try to infer them from the constructor call
+      if (!typeArgumentNodes && prototype.constructorPrototype && flow && ctxTypes.size == 0) {
```

Comment:
What's the `ctxTypes.size == 0` for?

Reply:
The else case was the logic that was there previously, and it handles inference for something like `let arr: Array<T> = new Array()` where the inference is from the type it's being assigned to.  Those are in `ctxTypes`.  So if they are present, then I need to _not_ infer from the parameters, but just use the previous logic.